source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -155,7 +157,8 @@ ProtobufUtil::Status JsonTranscoderConfig::createTranscoder(
   std::vector<VariableBinding> variable_bindings;
   method_descriptor =
       path_matcher_->Lookup(method, path, args, &variable_bindings, &request_info.body_field_path);
-  if (!method_descriptor) {
+  if (!method_descriptor ||
+      StringUtil::caseCompare(content_type, Http::Headers::get().ContentTypeValues.Grpc)) {
```

Comment:
Use `Grpc::Common::hasGrpcContentType(headers)` to check content-type, and in this case it can be returned at top of this function.